### ISSUE TITLE
What is the difference between tasks and workers？

### ISSUE THREAD

**Issue by canghaiyunfan** [huggingface/datatrove#195]
What is the difference between tasks and workers, what is the definition of tasks and how to determine the number of tasks?

**Comment by justHungryMan** [comment]
As a Datatrove user just passing by, I'd like to explain the difference between tasks and workers in the context of job execution.

Tasks refers to the division of a job into manageable sub-units. For instance, if you have 1 million data that need to be processed, you could divide this job into 10 tasks. Each task would then be responsible for handling 100,000 data.

On the other hand, workers refer to the number of these tasks that can be executed concurrently. Continuing with the previous example, if you decide to use 5 workers and have 10 tasks, this setup means that you can process five tasks at the same time. Consequently, each worker would handle one task at a time until all tasks are completed.

In summary, tasks define the subdivision of a job into smaller parts, while workers determine how many of these parts can be processed simultaneously.

**Comment by canghaiyunfan** [comment]
@justHungryMan Thanks!
